### ISSUE TITLE
🦠 Model Request: Drug Induced Liver Injury

### ISSUE THREAD

**Issue by GemmaTuron** [ersilia-os/ersilia#438]
### Model Title

DILI (TDC Dataset)

### Publication

Hello @DhanshreeA!

As part of your Outreachy contribution, we have assigned you the dataset "DILI" from the Therapeutics Data Commons to try and build a binary classification ML model.
Please copy the provided Google Colab template and use this issue to provide updates on the progress. We'll value not only being able to build the model but also interpreting its results.

**Comment by DhanshreeA** [comment]
Hi @GemmaTuron! I have successfully fetched the DILI data set from TDC. Following are the characteristics of the data set:

1. It is a divided into two classes - drugs that are toxic to the liver (induce injury), with target Y as 1.0, and drugs that are not toxic to the liver, with Y labeled as 0.0
2. The data set has 475 unique drugs in total. 
3. The data set is balanced, with 239 non toxic drugs and 236 toxic drugs.

I have drawn a toxic molecule `NC(=O)c1cnccn1` (which I believe is called [pyrazine-2-carboxamide](https://cactus.nci.nih.gov/chemical/structure/NC(=O)c1cnccn1/iupac_name)) and a non-toxic molecule `C[N+](C)(C)CC(=O)[O-]` ([2trimethylazaniumylacetate](https://cactus.nci.nih.gov/chemical/structure/%20C%5BN+%5D(C)(C)CC(=O)%5BO-%5D/iupac_name))
![index](https://user-images.githubusercontent.com/7213136/197852078-73eebe29-f3f0-4d27-ba1b-89ab547a700e.png)

I am a bit confused as to what further information can I plot to get a better understanding of the data. I was considering carrying out PCA on Morgan Fingerprints. Is that a good idea?

**Comment by DhanshreeA** [comment]
Hi @GemmaTuron! Thank you for your suggestions! I am still working on the PCA and iterating experimenting with a couple of other models for the dataset. 

These are the observations of the experiments so far. I have only trained a model using MorganBinaryClassifier from lazy-qsar. This has trained a `RandomForestClassifier` with 4 trees. 

Contingency table for the validation set:

![Screenshot from 2022-10-28 17-58-02](https://user-images.githubusercontent.com/7213136/198586916-a86ceffc-567d-435b-a75f-19cde0bcd6dc.png)

ROC Curve for the validation set:

![Screenshot from 2022-10-28 17-58-52](https://user-images.githubusercontent.com/7213136/198587084-0bd88ae7-7973-48cf-a397-d43514e9da69.png)

1. Precision score: 0.81
2. Recall score: 0.92
3. AUROC score: 0.88
4. F1 Score: 0.86
5. Accuracy: 0.83

Contingency table for test set:

![Screenshot from 2022-10-28 17-51-34](https://user-images.githubusercontent.com/7213136/198586480-fa2974f3-ea2a-4096-a6a3-da78b4dae983.png)

ROC Curve for the test set:

![Screenshot from 2022-10-28 17-56-23](https://user-images.githubusercontent.com/7213136/198586631-503e8cbf-f2bc-4328-88d1-bcdccbcbcda8.png)

1. Precision score: 0.74
2. Recall score: 0.771 (rounded)
3. AUROC score: 0.80
4. F1 Score: 0.75
6. Accuracy: 0.74

Contingency table for the training set:

![Screenshot from 2022-10-28 18-01-32](https://user-images.githubusercontent.com/7213136/198587477-52fc2c78-f02a-48a9-9843-5e3d7e533038.png)

1. Precision score: 0.78
2. Recall score: 0.81
4. F1 Score: 0.80
5. Accuracy:0.80

**Comment by DhanshreeA** [comment]
Addressing the above results, starting from the scores for the training set itself, we see that there is a some degree of variance in the model predictions. It is not accurately predicting the data it has already seen. In the scenario of Drug Induced Liver Injury, this is unacceptable, since the model is unable to classify about 19% (ie 30/161)  of the drugs that it has seen as toxic. This implies that if pharma companies were to solely rely on this model, 1 out of 5 drugs is likely to cause liver injuries during clinical trials on actual humans. One can only imagine the lawsuits that may follow such a situation!
A similar trend can be seen with 11 False negatives in the validation set with which is approximately 23% of the positive lables (ie toxic drugs) in the dataset. 
One interesting thing to note is that the accuracy measure is roughly the same as prescision across all the splits, and this is because the data set is fairly balanced.

**Comment by Femme-js** [comment]
I want to give a "Good Job' compliment to you @DhanshreeA for precisely describing the conclusion. I like the way you have put it out there.
